src/pu-package.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ */
+
+#include <fcntl.h>
+#include <gio/gio.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include "pu-error.h"
+#include "pu-package.h"
+#include "pu-utils.h"
+
+gboolean
+pu_package_create(GPtrArray *files,
+                  const gchar *output,
+                  gboolean force_overwrite,
+                  GError **error)
+{
+    g_autofree gchar *cmd = NULL;
+    g_autofree gchar *input = NULL;
+
+    g_return_val_if_fail(files != NULL, FALSE);
+    g_return_val_if_fail(g_strcmp0(output, "") > 0, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    for (guint i = 0; g_ptr_array_index(files, i) != NULL; i++) {
+        g_autofree gchar *file = g_strdup(g_ptr_array_index(files, i));
+        if (!g_file_test(file, G_FILE_TEST_EXISTS)) {
+            g_set_error(error, PU_ERROR, PU_ERROR_FAILED,
+                        "Input file '%s' does not exist", file);
+            return FALSE;
+        }
+    }
+
```

Comment:
We should check here if one of the input files is `layout.yaml` as the created package is invalid without it.